coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala

```diff
@@ -0,0 +1,237 @@
+package filodb.coordinator
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.Config
+import net.ceedubs.ficus.Ficus._
+import org.velvia.filo.RowReader
+import scala.concurrent.Future
+
+import filodb.core.metadata.{Column, Dataset, Projection, RichProjection}
+import filodb.core.columnstore.ColumnStore
+import filodb.core.reprojector.{MemTable, MapDBMemTable, Reprojector}
+
+object DatasetCoordinatorActor {
+  import filodb.core.Types._
+
+  /**
+   * Ingests a bunch of rows into this dataset, version's memtable.
+   * Automatically checks at the end if the memtable should be flushed, and sends a StartFlush
+   * message to itself to initiate a flush.  This is done before the Ack is sent back.
+   * @param rows the rows to ingest.  Must have the same schema, for now, as the columns passed in
+   *        when the ingestion was set up.
+   * @return Ack(seqNo)
+   */
+  case class NewRows(ackTo: ActorRef, rows: Seq[RowReader], seqNo: Long)
+
+  // The default minimum amt of memory in MB to allow ingesting more data
+  val DefaultMinFreeMb = 512
+
+  /**
+   * Initiates a memtable flush to the columnStore if one is not already in progress.
+   * Can also be used to listen to / wait for when a reprojection is finished.
+   * @param replyTo optionally, add to the list of folks to get a message back when flush/reprojection
+   *                finishes.
+   */
+  case class StartFlush(replyTo: Option[ActorRef] = None)
+
+  // Checks if memtable is ready to accept more rows.  Common reason why not is due to lack of memory
+  // or the row limit has been reached
+  case object CanIngest
+
+  /**
+   * Clears all data from the projection.  Waits for existing flush to finish first.
+   */
+  case class ClearProjection(replyTo: ActorRef, projection: Projection)
+
+  /**
+   * Returns current stats.  Note: numRows* will return -1 if the memtable is not set up yet
+   */
+  case object GetStats
+  case class Stats(flushesStarted: Int,
+                   flushesSucceeded: Int,
+                   flushesFailed: Int,
+                   numRowsActive: Int,
+                   numRowsFlushing: Int)
+
+  // Internal messages
+  case class FlushDone(result: Seq[String])
+  case class FlushFailed(t: Throwable)
```

Comment:
These case objects/classes need a sealed common trait to extend, that also explains the domain tree they fall under for FiloDB, perhaps that is Serializable